### PR TITLE
Heterogeneous null conformance tests

### PR DESCRIPTION
**Pull request by TristonianJones**
Add conformance tests for the null comparison related changes proposed and approved in https://github.com/google/cel-spec/wiki/proposal-210

### CODE REVIEW

File: tests/simple/testdata/comparisons.textproto [hunk 1]
```diff
@@ -204,6 +204,323 @@ section {
...
+    value: { bool_value: false }
+  }
+  test {
+    name: "not_eq_dyn_timestamp_null"
+    expr: "dyn(timestamp(0)) == null"
+    value: { bool_value: false }
+  }
+}
+section {
+  name: "eq_wrapper"
```
Comment (jnthntatum) [inline tests/simple/testdata/comparisons.textproto:271]: Could you add a bit more detail here or in the proposal about when the wrapper is treated as null (if the wrapper is a member and unset)? I think the distinction here isn't totally intuitive and I'm not sure when this would come up in practice?

**Review by jnthntatum — CHANGES_REQUESTED** [review]
LGTM other than adding a bit more about the protowrapper / null behavior cases.

File: tests/simple/testdata/comparisons.textproto (hunk 1 above)
Comment (TristonianJones) [reply tests/simple/testdata/comparisons.textproto:271]: I added tests to show the difference between unset, empty, and set. Wrapper types are tricky and not always implemented correctly in each of these cases.

**Comment by TristonianJones** [comment]
PTAL

**Review by jnthntatum — APPROVED** [review]
